R/analyse_functional_network.R

```diff
@@ -121,23 +121,60 @@ analyse_functional_network <- function(data,
     return(invisible(NULL))
   }
 
-  STRINGdb <- get("STRINGdb", envir = loadNamespace("STRINGdb"))
+  # Ensure data frame is not empty and columns exist
+  if (nrow(data) == 0) {
+    stop("The input data frame is empty.")
+  }
+
+  required_columns <- c(ensym(protein_id), ensym(string_id))
+  missing_columns <- required_columns[!required_columns %in% colnames(data)]
+
+  if (length(missing_columns) > 0) {
+    stop(
+      "The following required columns are missing from the input data frame: ",
+      paste(missing_columns, collapse = ", ")
+    )
+  }
+
+  if (plot && nrow(data) > 400) {
+    stop("Please only provide the top 400 significant proteins for plots! STRING cannot plot more at once.")
+  }
+
 
   data <- data %>%
     dplyr::distinct({{ protein_id }}, {{ string_id }}, {{ binds_treatment }})
 
   if (length(unique(dplyr::pull(data, !!ensym(protein_id)))) != nrow(data)) {
     stop(strwrap("Please provide unique annotations for each protein! The number of proteins
-does not match the number of rows in your data.", prefix = "\n", initial = ""))
+    does not match the number of rows in your data.", prefix = "\n", initial = ""))
   }
 
-  string_db <- STRINGdb$new(
-    version = version,
-    species = organism_id, # Check on String database to get the right code (E.coli K12: 511145)
-    score_threshold = score_threshold, # Cutoff score to consider something an interaction
-    input_directory = ""
+  if (!curl::has_internet()) {
+    message("No internet connection.")
+    return(invisible(NULL))
+  }
+
+  STRINGdb <- get("STRINGdb", envir = loadNamespace("STRINGdb"))
+
+  string_db <- tryCatch(
+    {
+      STRINGdb$new(
+        version = version,
+        species = organism_id, # Check on String database to get the right code (E.coli K12: 511145)
+        score_threshold = score_threshold, # Cutoff score to consider something an interaction
+        input_directory = ""
+      )
+    },
+    error = function(e) {
+      e$message
+    }
```

Comment:
Can you also catch the warning message here. You don't need to include the resulting message, but just not have it return the warning.

Reply:
should we print this part ("A warning occurred during the interaction network analysis: " or not?
 :)

Reply:
yes looks good like this actually! Thanks! :)